flink-runtime/src/main/java/org/apache/flink/runtime/security/token/DefaultDelegationTokenManager.java

```diff
@@ -45,91 +42,86 @@
 
     private final Configuration configuration;
 
     private final double tokensRenewalTimeRatio;
 
     private final long renewalRetryBackoffPeriod;
 
-    private final KerberosLoginProvider kerberosLoginProvider;
-
-    @VisibleForTesting final Map<String, HadoopDelegationTokenProvider> delegationTokenProviders;
+    @VisibleForTesting final Map<String, DelegationTokenProvider> delegationTokenProviders;
 
     @Nullable private final ScheduledExecutor scheduledExecutor;
 
     @Nullable private final ExecutorService ioExecutor;
 
-    @Nullable private ScheduledFuture<?> tgtRenewalFuture;
-
     private final Object tokensUpdateFutureLock = new Object();
 
     @GuardedBy("tokensUpdateFutureLock")
     @Nullable
     private ScheduledFuture<?> tokensUpdateFuture;
 
-    @Nullable private DelegationTokenListener delegationTokenListener;
+    @Nullable private Listener listener;
 
-    public KerberosDelegationTokenManager(
+    public DefaultDelegationTokenManager(
             Configuration configuration,
             @Nullable ScheduledExecutor scheduledExecutor,
             @Nullable ExecutorService ioExecutor) {
         this.configuration = checkNotNull(configuration, "Flink configuration must not be null");
-        this.tokensRenewalTimeRatio = configuration.get(KERBEROS_TOKENS_RENEWAL_TIME_RATIO);
+        this.tokensRenewalTimeRatio = configuration.get(DELEGATION_TOKENS_RENEWAL_TIME_RATIO);
         this.renewalRetryBackoffPeriod =
-                configuration.get(KERBEROS_TOKENS_RENEWAL_RETRY_BACKOFF).toMillis();
-        this.kerberosLoginProvider = new KerberosLoginProvider(configuration);
+                configuration.get(DELEGATION_TOKENS_RENEWAL_RETRY_BACKOFF).toMillis();
         this.delegationTokenProviders = loadProviders();
         this.scheduledExecutor = scheduledExecutor;
         this.ioExecutor = ioExecutor;
     }
 
-    private Map<String, HadoopDelegationTokenProvider> loadProviders() {
+    private Map<String, DelegationTokenProvider> loadProviders() {
         LOG.info("Loading delegation token providers");
 
-        ServiceLoader<HadoopDelegationTokenProvider> serviceLoader =
-                ServiceLoader.load(HadoopDelegationTokenProvider.class);
+        ServiceLoader<DelegationTokenProvider> serviceLoader =
+                ServiceLoader.load(DelegationTokenProvider.class);
 
-        Map<String, HadoopDelegationTokenProvider> providers = new HashMap<>();
-        for (HadoopDelegationTokenProvider provider : serviceLoader) {
+        Map<String, DelegationTokenProvider> providers = new HashMap<>();
+        for (DelegationTokenProvider provider : serviceLoader) {
             try {
                 if (isProviderEnabled(provider.serviceName())) {
                     provider.init(configuration);
                     LOG.info(
                             "Delegation token provider {} loaded and initialized",
                             provider.serviceName());
+                    checkState(
```

Comment:
What happens if multiple jars contain the `same` provider.
Or a shaded one? Or different versions of the same provider?

Reply:
If 2 providers are coming w/ the same name then it blows up and deployment fails fast.

Reply:
If we wouldn't do that then 2 providers would overwrite each others tokens. Race is never a good thing especially when it's hidden.

Reply:
I am fine with the solution, I was just interested in how this is solved in other places